presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -214,7 +219,33 @@ public void reserveMemory(long bytes)
     {
         ListenableFuture<?> future = driverContext.reserveMemory(bytes);
         if (!future.isDone()) {
-            memoryFuture.set(future);
+            SettableFuture<?> currentMemoryFuture = memoryFuture.get();
```

Comment:
add a comment that we can't replace an existing non-done future because the task maybe blocked on that future